zson/ztests/type-value.yaml

```
@@ -3,5 +3,6 @@ zed: "*"
 input: &input |
   <int64>
   {typ:<int64>}
+  <{a:"@foo"={bar:string},b:"@foo"}>
```

Nit: Different field names and a simpler underlying type would make the intent behind this case a little clearer.
```suggestion
  <{def:"@foo"=int64,ref:"@foo">
```